### PR TITLE
IR-9-Deleting-an-incident

### PR DESCRIPTION
**Pull request by Golder12**
The method deleteIncident() is passed an incident. The incident's id is retrieved. A loop through the list containing the saved incidents is created. The incident id we got is now checked against all saved Incidents. If a match is found, that corresponding incident is then deleted.

### CODE REVIEW

File: src/org/pahappa/systems/services/IncidentServiceImpl.java [hunk 1]
```diff
@@ -86,9 +86,22 @@ public Incident getIncidentOfId(int id) {
...
+		String incidentToDelete = incident.getTitle();
+		int incidentSize = incidents.size();
+
+		for (Incident item : incidents) {
+			if(incidentToDelete == item.getTitle()){
+				incidents.remove(item);
+				System.out.println("The incident has been successfully deleted");
+				break;
+			}
+		}
```
Comment (reifred) [inline src/org/pahappa/systems/services/IncidentServiceImpl.java]: You can solve this in a better way than iterating through the list to have a more readable code.

- Check if the incident given is not null
- Then make use `.remove` function on that list to delete that incident

**Review by reifred — CHANGES_REQUESTED** [review]

File: src/org/pahappa/systems/services/IncidentServiceImpl.java (hunk 1 above)
Comment (Golder12) [reply src/org/pahappa/systems/services/IncidentServiceImpl.java]: Ohhh, yes. that is better. Thank you.

Comment (Golder12) [reply src/org/pahappa/systems/services/IncidentServiceImpl.java]: I've updated the function

**Review by iandrew8 — APPROVED** [review]

**Review by katusiimeconrad — COMMENTED** [review]
I think you'll have to also check that the incident exists in the stored incidents.
The passed in parameter may not be null, but it doesn't exist in the stored incidents. @reifred does she have to throw an exception?

**Review by ttcollins — APPROVED** [review]

**Comment by ttcollins** [comment]
Besides checking for null, shouldn't there be a check for empty as well @Golder12

**Comment by reifred** [comment]
> I think you'll have to also check that the incident exists in the stored incidents.
> The passed in parameter may not be null, but it doesn't exist in the stored incidents. @reifred does she have to throw an exception?

@Golder12 update with this check too as suggested by @katusiimeconrad. To see if the incident exists.

**Comment by reifred** [comment]
> Besides checking for null, shouldn't there be a check for empty as well @Golder12

@ttcollins  an incident cannot be empty.

**Comment by Golder12** [comment]
> > I think you'll have to also check that the incident exists in the stored incidents.
> > The passed in parameter may not be null, but it doesn't exist in the stored incidents. @reifred does she have to throw an exception?
> 
> @Golder12 update with this check too as suggested by @katusiimeconrad. To see if the incident exists.

@reifred implemented a check to see if it exists.

**Review by fredmugerwa — CHANGES_REQUESTED** [review]
Kindly remove unwanted files
Screen Shot 2021-09-13 at 2 30 57 PM

**Comment by Golder12** [comment]
@reifred Don't merge this branch. It's corrupted. I'm making a new one for this ticket.

**Pull request closed without merging**